src/metpy/plots/declarative.py

```diff
@@ -928,8 +936,8 @@ def griddata(self):
             data_subset = data.metpy.sel(**subset).squeeze()
 
             if self.plot_units is not None:
-                data_subset = data_subset.metpy.convert_units(self.plot_units)
-            self._griddata = data_subset
+                data_subset.metpy.convert_units(self.plot_units)
```

Comment:
Since `convert_units` no-longer works in-place, should probably be:
```python
self._griddata =  data_subset.metpy.convert_units(self.plot_units) * self.scale
```